### PR TITLE
Add Dashboards integration workflow

### PR DESCRIPTION
**Pull request by BSFishy**
### Description
Adds a workflow to run tests in Dashboards with current OUI changes. This is meant to establish a pattern for how integrations could be run with Dashboards.

Also did some cleanup:
 - Split build and build docs jobs, so they can run in parallel
 - Put `name` at the top of all jobs to keep everything the same

### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] All tests pass
  - [x] `yarn lint`
  - [x] `yarn test-unit`
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/oui/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by AMoo-Miki — APPROVED** [review]

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -10,10 +10,15 @@ on:
     paths-ignore:
       - '**/*.md'
 
+env:
+  CI: 1
+  GCS_UPLOAD_PREFIX: fake
```
Comment (joshuarrrr) [inline .github/workflows/build_and_test.yml:15]: curiosity - what does this do?

**Review by joshuarrrr — APPROVED** [review]
This is awesome! Hopefully we don't add too much time or false positives to our OUI PRs, but I think this should be a big win.

File: .github/workflows/build_and_test.yml (hunk 1 above)
Comment (BSFishy) [reply .github/workflows/build_and_test.yml:15]: Not completely sure. When I ran it without this var, one of the tests failed, so I just copied it from the OSD build and test workflow

**Comment by BSFishy** [comment]
> This is awesome! Hopefully we don't add too much time or false positives to our OUI PRs, but I think this should be a big win.

Exactly. This should ideally push down the number of false positives we have, as it adds an automated check to see if OUI integrated into OSD effectively (i.e. we're not breaking anything), so we can catch issues like that when the PR is made, rather than 3 days before code freeze.

**Pull request merged**